pkg/project/manager.go

```diff
@@ -218,75 +186,108 @@ func (pm ManagerImpl) DeleteProject(projectId string) <-chan result.Empty {
 
 	project, err := pm.GetProject(projectId)
 
 	if err != nil {
-		log.Printf("Project with id %s not found", projectId)
+		log.Error().Err(err).Msgf("Project with id %s not found", projectId)
 		return TaskResult{}, fmt.Errorf("Project with id %s not found", projectId)
 	}
 
-	execResult, err := pm.DevContainerRunner.Exec(project.containerId, []string{"/bin/bash", "-c", command})
+	execResult, err := pm.DevContainerRunner.Exec(project.ContainerId, []string{"/bin/bash", "-c", command})
 
 	if err != nil {
-		log.Printf("Failed to execute command '%s' in container %s: %s", command, project.containerId, err)
+		log.Error().Err(err).Msgf("Failed to execute command '%s' in container %s", command, project.ContainerId)
 		return TaskResult{}, fmt.Errorf("Failed to execute command: %w", err)
 	}
 
-	log.Printf("Task '%s' for project %s completed", command, projectId)
+	log.Debug().Msgf("Task '%s' for project %s completed", command, projectId)
 
 	return TaskResult{StdOut: execResult.StdOut, StdErr: execResult.StdErr, ExitCode: execResult.ExitCode}, nil
 }
 
 func (pm ManagerImpl) Cleanup() error {
-	log.Printf("Cleaning up projects")
+	log.Info().Msg("Cleaning up projects")
 
 	projects, err := pm.GetProjects()
 
 	if err != nil {
-		log.Printf("Failed to get projects: %s", err)
+		log.Error().Err(err).Msg("Failed to get projects")
 		return fmt.Errorf("Failed to get projects: %w", err)
 	}
 
 	for _, project := range projects {
-		log.Printf("Cleaning up project %s", project.Id)
-		pm.DevContainerRunner.Stop(project.containerId)
+		log.Debug().Msgf("Cleaning up project %s", project.Id)
+		pm.DevContainerRunner.Stop(project.ContainerId)
 	}
 
-	log.Printf("Cleaned up projects")
+	log.Info().Msg("Cleaned up projects")
 
 	return nil
 }
 
+func (pm ManagerImpl) CreateFile(ctx context.Context, projectId, path, content string) (model.File, error) {
+	log.Debug().Msgf("Creating file %s in project %s", path, projectId)
+
+	project, err := pm.GetProject(projectId)
+
+	if err != nil {
+		log.Error().Err(err).Msgf("Project with id %s not found", projectId)
+		return model.File{}, fmt.Errorf("Project with id %s not found", projectId)
+	}
+
+	return pm.fileManager.CreateFile(model.NewContextWithProject(ctx, &project), afero.NewBasePathFs(afero.NewOsFs(), project.Path), path, content)
+}
+
+func (pm ManagerImpl) ReadFile(ctx context.Context, projectId, path string, props files.ReadProps) (model.File, error) {
+	project, err := pm.GetProject(projectId)
+
+	if err != nil {
+		return model.File{}, fmt.Errorf("Project with id %s not found", projectId)
+	}
+
+	return pm.fileManager.ReadFile(model.NewContextWithProject(ctx, &project), afero.NewBasePathFs(afero.NewOsFs(), project.Path), path, props)
+}
+
+func (pm ManagerImpl) UpdateFile(ctx context.Context, projectId, path, content string) (model.File, error) {
+	project, err := pm.GetProject(projectId)
+
+	if err != nil {
+		return model.File{}, fmt.Errorf("Project with id %s not found", projectId)
```

Comment:
Here, the error does not necessarily mean not found, it can be anything else. We need to distinguish between project not found and not being able to fetch it (due to network issues for example). One option is to return a pointer and to check if it's nil.